datanode/broker/buffered_event_source.go

```diff
@@ -270,14 +301,41 @@ func (m *FileBufferedEventSource) removeBufferFile(bufferFile *os.File) error {
 		return fmt.Errorf("failed to close last event buffer file:%w", err)
 	}
 
-	err = os.Remove(bufferFile.Name())
+	if m.config.Archive {
+		err = m.moveBufferFileToArchive(bufferFile.Name())
+		if err != nil {
+			return fmt.Errorf("failed to move buffer file to archive: %w", err)
+		}
+	} else {
+		err = os.Remove(bufferFile.Name())
+		if err != nil {
+			return fmt.Errorf("failed to remove event buffer file: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// moveBufferFileToArchive encodes the creation time into the archive file name to ensure that the correct order
+// of files can always be determined even if the archive files are copied etc.
+func (m *FileBufferedEventSource) moveBufferFileToArchive(bufferFilePath string) error {
+	bufferFileName := filepath.Base(bufferFilePath)
+	bufferSeqSpan := strings.ReplaceAll(bufferFileName, bufferFileNamePrepend, "")
+	timeNowUtc := time.Now().UTC()
+
+	archiveFileName := fmt.Sprintf("%s-%s-%d-seqnumspan%s", bufferFileNamePrepend,
+		timeNowUtc.Format("2006-01-02-15:04:05"), timeNowUtc.UnixNano(), bufferSeqSpan)
```

Comment:
I don't think having colons in file names are a good idea. It's a bit messy to escape the characters.

Reply:
good point - sorted